download/config.go

```diff
@@ -23,39 +25,54 @@ type PollingConfig struct {
 
 // Config represents the configuration for the downloader.
 type Config struct {
+	Trigger *string       `json:"trigger,omitempty"`
 	Polling PollingConfig `json:"polling"`
 }
 
 // ValidateAndInjectDefaults checks for configuration errors and ensures all
 // values are set on the Config object.
 func (c *Config) ValidateAndInjectDefaults() error {
 
-	min := defaultMinDelaySeconds
-	max := defaultMaxDelaySeconds
+	if c.Trigger == nil {
+		t := plugins.DefaultTriggerMode
+		c.Trigger = &t
+	}
 
-	// reject bad min/max values
-	if c.Polling.MaxDelaySeconds != nil && c.Polling.MinDelaySeconds != nil {
-		if *c.Polling.MaxDelaySeconds < *c.Polling.MinDelaySeconds {
-			return fmt.Errorf("max polling delay must be >= min polling delay")
-		}
-		min = *c.Polling.MinDelaySeconds
-		max = *c.Polling.MaxDelaySeconds
-	} else if c.Polling.MaxDelaySeconds == nil && c.Polling.MinDelaySeconds != nil {
-		return fmt.Errorf("polling configuration missing 'max_delay_seconds'")
-	} else if c.Polling.MinDelaySeconds == nil && c.Polling.MaxDelaySeconds != nil {
-		return fmt.Errorf("polling configuration missing 'min_delay_seconds'")
+	switch *c.Trigger {
+	case plugins.TriggerPeriodic, plugins.TriggerManual:
+		break
```

Comment:
[nit] Does this have a different effect than not doing anything? 🤔 
```suggestion
		// nothing to do
```